packages/amplify-ui-components/src/components/amplify-require-new-password/amplify-require-new-password.tsx

```diff
@@ -0,0 +1,109 @@
+import { Component, Prop, State, h } from '@stencil/core';
+import { FormFieldTypes } from '../amplify-auth-fields/amplify-auth-fields-interface';
+import { AuthState, ChallengeName, CognitoUserInterface } from '../../common/types/auth-types';
+import { CHANGE_PASSWORD, CHANGE, BACK_TO_SIGN_IN, NO_AUTH_MODULE_FOUND } from '../../common/constants';
+// import { CodeDeliveryType } from './amplify-forgot-password-interface';
+
+import { Auth } from '@aws-amplify/auth';
+import { Logger, isEmpty } from '@aws-amplify/core';
+
+const logger = new Logger('amplify-require-new-password');
+
+@Component({
+  tag: 'amplify-require-new-password',
+  shadow: false,
+})
+export class AmplifyRequireNewPassword {
+  /** The header text of the forgot password section */
+  @Prop() headerText: string = CHANGE_PASSWORD;
+  /** The text displayed inside of the submit button for the form */
+  @Prop() submitButtonText: string = CHANGE;
+  /** The form fields displayed inside of the forgot password form */
+  @Prop() formFields: FormFieldTypes;
+  /** (Optional) Overrides default styling */
+  @Prop() overrideStyle: boolean = false;
+  /** The function called when submitting a new password */
+  @Prop() handleSubmit: (event: Event) => void = event => this.change(event);
+  /** Passed from the Authenticatior component in order to change Authentication state */
+  @Prop() handleAuthStateChange: (nextAuthState: AuthState, data?: object) => void;
```

Comment:
Do you think it's time we abstract this into one of the auth types?

```ts
type AuthStateHandler = (nextAuthState: AuthState, data?: object) => void;
```

```suggestion
  @Prop() handleAuthStateChange: AuthStateHandler;
```

Reply:
I think this is a good idea. I can make a follow-up PR for this.